core/src/test/java/org/infinispan/hadoop/FailoverHandlingTest.java

```diff
@@ -0,0 +1,300 @@
+package org.infinispan.hadoop;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.infinispan.arquillian.core.HotRodEndpoint;
+import org.infinispan.arquillian.core.InfinispanResource;
+import org.infinispan.arquillian.core.RemoteInfinispanServer;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.filter.KeyValueFilterConverterFactory;
+import org.infinispan.hadoop.testutils.Utils;
+import org.infinispan.hadoop.testutils.converters.CustomFilterFactory;
+import org.infinispan.hadoop.testutils.converters.ToWritableOutputConverter;
+import org.infinispan.hadoop.testutils.domain.CategoryStats;
+import org.infinispan.hadoop.testutils.domain.WebPage;
+import org.infinispan.hadoop.testutils.hadoop.MiniHadoopCluster;
+import org.infinispan.hadoop.testutils.mapreduce.CacheMapper;
+import org.infinispan.hadoop.testutils.mapreduce.HDFSReducer;
+import org.infinispan.hadoop.testutils.mapreduce.InfinispanMapper;
+import org.infinispan.hadoop.testutils.mapreduce.InfinispanReducer;
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OverProtocol;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Testing the failover cases, i.e. the proper work of Hadoop MapReduce in case if one of the clustered servers will
+ * go down.
+ *
+ * @author amanukya
+ */
+@RunWith(Arquillian.class)
+public class FailoverHandlingTest {
+
+    private static final String INPUT_CACHE_NAME = "default";
+    private static final String OUTPUT_CACHE_NAME = "namedCache";
```

Comment:
This cache was removed from Infinispan 8.2.0.CR1 so we need to use "memcachedCache" for now